nimbus/src/enrollment.rs

```diff
@@ -124,7 +133,7 @@ pub fn update_enrollments(
                 log::debug!("Experiment '{}' already has enrollment {:?}", slug, enr)
             }
             (Some(exp), None) => {
-                let enr = evaluate_enrollment(nimbus_id, aru, app_context, &exp)?;
+                let enr = evaluate_enrollment(nimbus_id, aru, app_context, &exp, schema_version)?;
```

Comment:
If `evaluate_enrollment` returns an error, IIUC we're going to immediately bail out and return an error from `update_enrollments`, which will in turn fail the call to `NimbusClient.update_experiments`. That might cause us to not be enrolled in other, perfectly valid experiments because we bailed out of the loop early.

I know `evaluate_enrollment` was already fallible so we kind of already had this problem, but I think the introduction of the new `SchemaError` could make it much more likely to happen in practice. Encountering an experiment with an unexpected schema doesn't feel like it should be a fatal error.